llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -6111,6 +6141,12 @@ void BoUpSLP::buildTree_rec(ArrayRef<Value *> VL, unsigned Depth,
     if (NumUniqueScalarValues == VL.size()) {
       ReuseShuffleIndicies.clear();
     } else {
+      if (UserTreeIdx.UserTE && UserTreeIdx.UserTE->isNonPowOf2Vec()) {
```

Comment:
Add FIXME here for non-power-of-2 support

Reply:
Added, thanks!